### DISCUSSION TITLE
How does SelectionPolygonVolume work?

### DISCUSSION THREAD

**Discussion by guhur** [isl-org/Open3D#4193] [Q&A]
Hi,

I am surprised by the following result:

```python
points = np.array([[-0.37841943, -0.05783039,  2.42544341],
       [-0.72100592,  0.5099898 ,  1.67696834],
       [ 0.13754743,  1.0226984 ,  1.67295718],
       [ 0.48013398,  0.45487815,  2.42143226],
       [ 0.26103601, -0.16154499,  1.38604999]])
pcd = o3d.geometry.PointCloud()
pcd.points = o3d.utility.Vector3dVector(points)
print(pcd)

selector = open3d.visualization.SelectionPolygonVolume()
selector.bounding_polygon = o3d.utility.Vector3dVector(points)

selected = selector.crop_point_cloud(pcd)
print(selected)
```

It prints:

> PointCloud with 5 points.
> PointCloud with 0 points.

So basically, cropping a Point Cloud with itself returns nothing (!?)

Can you enlighten me about it is going on?

**Comment by mserouar** [comment]
Same there, including after `.astype("float64")` tranformation.

The fact is that you need to specify what axis to orient the polygon vertices to.

Here for example the "X" axis. 

```
vol.axis_max = np.max(bounding_polygon[:, 0])
vol.axis_min = np.min(bounding_polygon[:, 0])
```

**Comment by veenusav** [comment]
vol = o3d.visualization.SelectionPolygonVolume()
    vol.orthogonal_axis = "Z"
    vol.axis_min = -2000
    vol.axis_max = 4000
    vol.bounding_polygon =  o3d.utility.Vector3dVector(points)
    cropped = vol.crop_point_cloud(pcd)
    
    the above kind of code worked.
